File: frontend/src/app/domain/objects/simplified-playlist-object.ts
import {ExternalURLObject} from './external-urlobject';
import {ImageObject} from './image-object';
import {OwnerObject} from './owner-object';
import {FollowersObject} from './followers-object';

export interface SimplifiedPlaylistObject {
  collaborative: boolean;
  description: string;
  external_urls: ExternalURLObject;
  href: string;
  id: string;
  images: ImageObject[];
  name: string;
  owner: OwnerObject;
  public: boolean;
  snapshot_id: string;
  tracks: FollowersObject;
  type: string;
  uri: string;
}

File: backend/src/main/java/organizer/backend/domain/objects/SimplifiedPlaylistObject.java
package organizer.backend.domain.objects;

import com.fasterxml.jackson.annotation.JsonProperty;

public record SimplifiedPlaylistObject(Boolean collaborative,
                                       String description,
                                       ExternalURLObject external_urls,
                                       String href,
                                       String id,
                                       ImageObject[] images,
                                       String name,
                                       OwnerObject owner,
                                       @JsonProperty("public") Boolean publicValue,
                                       String snapshot_id,
                                       FollowersObject tracks,
                                       String type,
                                       String uri) {
}
